test/framework/functional/framework.go

```diff
@@ -181,10 +181,8 @@ func (f *CollectorFunctionalFramework) DeployWithVisitors(visitors []runtime.Pod
 		if f.Conf, err = forwarder.Generate(logging.LogCollectionType(f.collector.String()), string(clfYaml), false, debugOutput, testClient); err != nil {
 			return err
 		}
-		//remove journal for now since we can not mimic it
-		re := regexp.MustCompile(`(?msU).*<source>.*type.*systemd.*</source>`)
-		f.Conf = string(re.ReplaceAll([]byte(f.Conf), []byte{}))
-
+		//mock sources to facilitate functional testing
+		f.Conf = f.collector.ModifyConfig(f.Conf)
```

Comment:
does this mean every functional test will replace journal logs with fake journal logs?

Reply:
Yes. This is a hook to allow the framework to modify a generated configuration.  This allows us to replace journal with something we can control as part of the framework.